include/exadg/poisson/overset_grids/user_interface/application_base.h

```diff
@@ -90,6 +90,77 @@ locate_object_relative_to_triangulation(
     return Location::Intersected;
 }
 
+template<int dim>
+void
+set_boundary_ids_overlap_region(dealii::Triangulation<dim> const & tria_dst,
+                                dealii::types::boundary_id const & bid,
+                                dealii::Mapping<dim> const &       mapping_src,
+                                dealii::Triangulation<dim> const & tria_src)
+{
+  std::vector<dealii::Point<dim>> points;
+  using CellIteratorType = decltype(tria_dst.begin());
+  using Id               = std::tuple<CellIteratorType /* cell */, unsigned int /*face*/>;
+  std::vector<std::pair<Id, unsigned int /* first_point_in_vector */>> id_to_vector_index;
+
+  // fill vector of points for all boundary faces
+  for(auto cell : tria_dst.cell_iterators())
+  {
+    for(auto const & f : cell->face_indices())
+    {
+      if(cell->face(f)->at_boundary())
+      {
+        for(auto const & v : cell->face(f)->vertex_indices())
+        {
+          if(v == 0)
+          {
+            Id const id = std::make_tuple(cell, f);
+            id_to_vector_index.push_back({id, points.size()});
+          }
+
+          points.push_back(cell->face(f)->vertex(v));
+        }
+      }
+    }
+  }
+
+  // create and reinit RemotePointEvaluation: find points on src-side
+  std::vector<bool> marked_vertices = {};
+  double const      tolerance       = 1.e-10;
+
+  dealii::Utilities::MPI::RemotePointEvaluation<dim> rpe =
+    dealii::Utilities::MPI::RemotePointEvaluation<dim>(tolerance,
+                                                       false
+#if DEAL_II_VERSION_GTE(9, 4, 0)
+                                                       ,
+                                                       0,
+                                                       [marked_vertices]() {
+                                                         return marked_vertices;
+                                                       }
+#endif
+    );
+
+  rpe.reinit(points, tria_src, mapping_src);
+
+  // check which points have been found and whether a face on dst-side is located inside the src
+  // triangulation
+  for(auto iter = id_to_vector_index.begin(); iter != id_to_vector_index.end(); ++iter)
+  {
+    unsigned int const begin = iter->second;
+    unsigned int const end =
+      (iter + 1 != id_to_vector_index.end()) ? (iter + 1)->second : points.size();
```

Comment:
I would split up `id_to_vector_index` into to vectors and insert in the vector with the pointers also `points.size()` as is common in the case of CRS.

Reply:
I don't understand what you mean here. Would you like to suggest this as a follow-up?

Reply:
Let's go with the current solution.